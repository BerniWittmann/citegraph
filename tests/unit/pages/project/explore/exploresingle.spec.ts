import { shallowMount } from '@vue/test-utils'
import { i18n } from '../../../setupPlugins'

import ExploreSinglePage from '@/pages/project/explore/ExploreSingle.vue'
import RecordViewPage from '@/components/project/explore/view/Record.vue'
import AuthorViewPage from '@/components/project/explore/view/Author.vue'

describe('pages/project/ExploreSinglePage.vue', () => {
  function getWrapper (queryByType: string = 'record') {
    return shallowMount(ExploreSinglePage, {
      i18n,
      mocks: {
        $route: {
          params: {
            queryByType
          }
        }
      }
    })
  }

  it('renders the page', () => {
    const wrapper = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('handles an invalid type', () => {
    const wrapper = getWrapper('invalid')
    expect(wrapper.html()).toMatchSnapshot()
  })

  const components: Record<string, any> = {
    record: RecordViewPage,
    author: AuthorViewPage
  }

  describe('it renders the correct component', () => {
    Object.keys(components).forEach((type) => {
      describe(type, () => {
        it('renders the component', () => {
          const wrapper = getWrapper(type)
          expect(wrapper.contains(components[type])).toBeTruthy()
        })
      })
    })
  })
})
