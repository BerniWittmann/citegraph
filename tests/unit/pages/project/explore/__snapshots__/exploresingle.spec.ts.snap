// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/project/ExploreSinglePage.vue handles an invalid type 1`] = `
<div>
  <!---->
</div>
`;

exports[`pages/project/ExploreSinglePage.vue renders the page 1`] = `
<div>
  <recordviewpage-stub></recordviewpage-stub>
</div>
`;
